rust/operator-binary/src/history/history_controller.rs

```diff
@@ -701,13 +725,51 @@ fn env_vars(s3logdir: &S3LogDir) -> Vec<EnvVar> {
         ]);
     }
 
-    vars.push(EnvVar {
-        name: "SPARK_HISTORY_OPTS".to_string(),
-        value: Some(history_opts.join(" ")),
-        value_from: None,
-    });
-    // if TLS is enabled build truststore
-    vars
+    vars.insert(
+        "SPARK_HISTORY_OPTS".to_string(),
+        EnvVar {
+            name: "SPARK_HISTORY_OPTS".to_string(),
+            value: Some(history_opts.join(" ")),
+            value_from: None,
+        },
+    );
+
+    // apply the role overrides
+    let mut role_envs = role_env_overrides
+        .into_iter()
+        .map(|env_var| {
+            (
+                env_var.0.clone(),
+                EnvVar {
+                    name: env_var.0.clone(),
+                    value: Some(env_var.1),
+                    value_from: None,
+                },
+            )
+        })
+        .collect();
+
+    vars.append(&mut role_envs);
```

Comment:
Just FYI, no need to change anything: There is also `extend`, which takes an iterator, so you don't need to `.collect()` `role_envs` before

Reply:
Thanks! Done 248479d

Reply:
Thanks as well!